### PR TITLE
Port visualizations to ROS2

### PR DESCRIPTION
**Pull request by Sarath18**
Visualizations ported to ROS2:

- PointStamped (`geometry_msgs/msg/PointStamped`)
- PoseStamped (`geometry_msgs/msg/PoseStamped`)
- Marker (`visualization_msgs/msg/Marker`)
- Imu (`sensor_msgs/msg/Imu`)
- static_tf (`tf2_msgs/msg/TFMessage`)
- PointCloud2 (`sensor_msgs/msg/PointCloud2`)

### CODE REVIEW

**Comment by Sarath18** [comment]
I have ported all the visualization to ROS2 @ahcorde can you please review the changes?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,27 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(ign-rviz)
+add_compile_options(-std=c++14 -O2)
```
Comment (ahcorde) [inline CMakeLists.txt]: why are you adding the flag -02 ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+set(PYTHON_EXECUTABLE "/usr/bin/python3")
```
Comment (ahcorde) [inline CMakeLists.txt]: Please use the right Cmake functions to find python3

File: include/ign-rviz/VisualizationManager.h [hunk 3]
```diff
@@ -33,29 +34,29 @@
...
   MarkerPtr pcl_marker;
 
-  tf2_ros::Buffer tfBuffer;
-  tf2_ros::TransformListener tfListener;
+  std::shared_ptr<tf2_ros::Buffer> tfBuffer;
+  std::shared_ptr<tf2_ros::TransformListener> tfListener;
 
  public:
-  VisualizationManager(int &argc, char** argv);
-  void point_callback(const geometry_msgs::PointStampedConstPtr&);
```
Comment (ahcorde) [inline include/ign-rviz/VisualizationManager.h]: can you add docblock? here an example https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/clock.hpp#L58

File: scripts/marker_TEST.py [hunk 4]
```diff
@@ -1,12 +1,11 @@
-import rospy
 from visualization_msgs.msg import Marker
 from geometry_msgs.msg import Point
 
 
 def create_line(marker_id, ns, marker_type):
     marker = Marker()
 
-    marker.header.stamp = rospy.Time().now()
+    # marker.header.stamp = rospy.Time().now()
```
Comment (ahcorde) [inline scripts/marker_TEST.py]: Please use the right method

File: scripts/marker_TEST.py [hunk 5]
```diff
@@ -15,99 +14,100 @@ def create_line(marker_id, ns, marker_type):
...
+    marker.scale.z = 1.0
 
     return marker
 
 
 def create_marker(marker_id, ns, marker_type):
     marker = Marker()
 
-    marker.header.stamp = rospy.Time().now()
+    # marker.header.stamp = rospy.Time().now()
```
Comment (ahcorde) [inline scripts/marker_TEST.py]: use the right method

File: scripts/marker_TEST.py [hunk 6]
```diff
@@ -15,99 +14,100 @@ def create_line(marker_id, ns, marker_type):
...
-    marker.scale.z = 1
+    marker.scale.x = 1.0
+    marker.scale.y = 1.0
+    marker.scale.z = 1.0
 
     return marker
 
 
 if __name__ == "__main__":
+    import rospy
```
Comment (ahcorde) [inline scripts/marker_TEST.py]: import the library in the top of the file

File: scripts/marker_TEST.py [hunk 7]
```diff
@@ -15,99 +14,100 @@ def create_line(marker_id, ns, marker_type):
...
     rospy.init_node("marker_test_node")
     marker_pub = rospy.Publisher("/visualization_marker", Marker, queue_size=10)
     rate = rospy.Rate(1)
```
Comment (ahcorde) [inline scripts/marker_TEST.py]: you are declaring the publisher but you are never using it

Comment (ahcorde) [inline scripts/marker_TEST.py]: unused

**Comment by ahcorde** [comment]
Please add the common linters to run: lint_cmake, uncrustify, cppcheck and others

Some pointers:

https://github.com/ros2/demos/blob/master/demo_nodes_cpp/package.xml#L32

https://github.com/ros2/demos/blob/master/demo_nodes_cpp/CMakeLists.txt#L140

File: CMakeLists.txt (hunk 1 above)
Comment (Sarath18) [reply CMakeLists.txt]: Initially, when I was writing the code for PointCloud2 visualization, it was too slow. I added compiler optimization to check if there is any significant boost in the code. I'll remove this if not required.

File: CMakeLists.txt (hunk 2 above)
Comment (Sarath18) [reply CMakeLists.txt]: Ok

File: include/ign-rviz/VisualizationManager.h (hunk 3 above)
Comment (Sarath18) [reply include/ign-rviz/VisualizationManager.h]: Sure

**Comment by Sarath18** [comment]
I mixed up the text code for ROS and ROS2, hence the inconsistency. I'll update the test script along with the above-suggested changes as soon as possible.

**Comment by Sarath18** [comment]
I have made the suggested changes. Can you please review the changes?

**Comment by ahcorde** [comment]
I think we can close this one

**Pull request closed without merging**